uvm/impl/com/untangle/uvm/DeviceTableImpl.java

```diff
@@ -190,56 +190,56 @@ public synchronized DeviceTableEntry addDevice(String macAddress)
-        logger.warn("lookupMacVendor:" + macAddress);
-        if (macAddress == null) return null;
-
+        logger.warn("lookupMacVendor:" + macAddresses);
+        if (macAddresses == null  ||  macAddresses.isEmpty()  && macAddresses.isBlank()) return null;
+        String[] macAddressList = macAddresses.split(",");
         try {
-            // forms the body to fetch the vendors
-            String body = "[" + macAddress + "]";
-            logger.info("Cloud MAC lookup = " + macAddress);
-            // forming the URL
-            URL myurl = new URL(UvmContextFactory.context().uriManager().getUri(CLOUD_LOOKUP_URL));
-            HttpURLConnection mycon;
-            if(myurl.getProtocol().equals("https")){
-                mycon = (HttpsURLConnection) myurl.openConnection();
-            }else{
-                mycon = (HttpURLConnection) myurl.openConnection();
-            }
-            mycon.setRequestMethod("POST");
-            mycon.setConnectTimeout(5000); // 5 seconds
-            mycon.setReadTimeout(5000); // 5 seconds
-            mycon.setRequestProperty("Content-length", String.valueOf(body.length()));
-            mycon.setRequestProperty("Content-Type", "application/json");
-            mycon.setRequestProperty("User-Agent", "Untangle NGFW Device Table");
-            mycon.setRequestProperty("AuthRequest", CLOUD_LOOKUP_KEY);
-            mycon.setDoOutput(true);
-            mycon.setDoInput(true);
-
-            DataOutputStream output = new DataOutputStream(mycon.getOutputStream());
-            output.writeBytes(body);
-            output.close();
-
-            DataInputStream input = new DataInputStream(mycon.getInputStream());
-            StringBuilder builder = new StringBuilder(256);
-
-            for (int c = input.read(); c != -1; c = input.read()) {
-                builder.append((char) c);
-            }
-
-            input.close();
-            mycon.disconnect();
+            if (macAddressList != null) {
+                JSONArray macAdressList = new JSONArray(macAddressList);
+                logger.info("Cloud MAC lookup = " + macAdressList);
+                
+                String body = macAdressList.toString();
+                
+                // forming the URL
+                URL myurl = new URL(UvmContextFactory.context().uriManager().getUri(CLOUD_LOOKUP_URL));
+                HttpURLConnection mycon;
+                if(myurl.getProtocol().equals("https")){
+                        mycon = (HttpsURLConnection) myurl.openConnection();
+                }else{
+                        mycon = (HttpURLConnection) myurl.openConnection();
+                }
+                mycon.setRequestMethod("POST");
+                mycon.setConnectTimeout(5000); // 5 seconds
+                mycon.setReadTimeout(5000); // 5 seconds
+                mycon.setRequestProperty("Content-length", String.valueOf(body.length()));
+                mycon.setRequestProperty("Content-Type", "application/json");
```

Comment:
Not sure how frequently this function is going to get called, but if it is, better to use constants for all strings. Check if we already have defined string constants or if any java library has defined at least the standard string constants like content-type, https, application/json.

Reply:
This method is called during network synchronization, when adding devices, or when refreshing the network tab. We can use Java's HttpClient library to provide constants for Content-Type and User-Agent, but still we will need to create our own string constants for other values.
Additionally, switching to the HttpClient library will require library import and need to update the connection logic to align with its usage, so I am planning to use string constants for all HTTP headers and related values.
let me know your thought. will do changes accordingly.